### ISSUE TITLE
Draggable points: unexpected behaviors when trying to drag columns

### ISSUE THREAD

**Issue by raf18seb** [highcharts/highcharts#15249]
Few concerns about draggable columns. For now, I'm listing them in one common ticket.

1. I would like to be able to drag the 1st (blue) column but nothing happens when I'm trying. To drag the blue column between the dark and green ones, I need to first do a series of other drag moves which is very unintuitive.
![dragging-first-column](https://user-images.githubusercontent.com/23310742/109789741-ed732780-7c10-11eb-8311-e38aea1c687a.gif)

2. It's possible to drag the 2nd (dark) column over the 1st column, but very hard to drag it over the 3rd one (green). Again, to walk around it, I need to do a series of the following steps:
![dragging-second-column](https://user-images.githubusercontent.com/23310742/109790876-219b1800-7c12-11eb-95c8-e8e0b43483c0.gif)

3. Turn on stacking. Drag the 2nd column over the first one - works fine. Now try to drag the 3rd column over the first one.

#### Live demo with steps to reproduce
https://jsfiddle.net/BlackLabel/e04om7wc/

The 3rd issue is solved when we define points in separate series: https://jsfiddle.net/BlackLabel/mbxf79yn/

#### Product version
Highcharts 9.0.1

**Comment by KacperMadej** [comment]
Dragging the 1st point changes its x value as expected and categories for xAxis are calculated again because nothing stops them. See with tooltip how all points have new x values calculated based on the lack of set x for the points in here: https://jsfiddle.net/BlackLabel/e04om7wc/1/
If you set x for points then the 1st point drags fine: https://jsfiddle.net/BlackLabel/e04om7wc/2/
If you want to drop a point between categories then `dragPrecisionX` needs to be lower: https://jsfiddle.net/BlackLabel/e04om7wc/3/

I would say that behavior is logical and expected (to oversimplify: categories for axis are calculated from points names and x values are calculated from categories) while on the other hand stacking within the same series is not fully supported (it is just not supported - it works to some point but not by design).

Probably not, but if you want to drag a point and drop on another and expect a switch between the points then maybe there should be an enhancement for this and it could be done with an extra key pressed (like shift or alt).

2 and 3 are more or less the same as 1 because setting x for points fixes them - https://jsfiddle.net/BlackLabel/e04om7wc/4/ 

tl;dr Stacking within the same series isn't supported. Setting x for points fixes all.

**Comment by rkamysz** [comment]
Hi @KacperMadej @raf18seb, `dragPrecisionX/Y` somehow improves the behavior of dragNdrop but still requires the user to shuffle the rest of the objects to get the desired effect. 

https://user-images.githubusercontent.com/9402370/109828668-23c69c00-7c3d-11eb-9096-5da382995195.mov

Due to the changed values the graph also changed (logical) but I just wanted to move element **1** to the middle and keep the previous space between them. The problem is getting bigger when I have more than 3 of them.

Unfortunately, I can see how customers will complain about it. 
Maybe ... assuming you have `dragPrecisionX / Y` set to 1, add another property `gap/padding/space...` (also with precision) that would cause different behaviour:
- item dropped between two other items will be squeezed into this place (changing x position by 1 of all items before and after in range `0...num of items`)
- if item is dropped directly onto another, they swap places

**Comment by KacperMadej** [comment]
Hi @rkamysz 
This seems like a good idea for an enhancement suggestion - feel free to open a [new ticket](https://github.com/highcharts/highcharts/issues/new?assignees=&labels=Type%3A+Feature+Request&template=feature_request.md&title=) for it with full description.
Here's a PoC (based on existing API, so you could as well use it already in the current Highcharts version): https://jsfiddle.net/BlackLabel/e04om7wc/6/
